File: documentation/Exercise 1/Java/src/main/java/de/thu/recipebook/activities/RecipeDetailsActivity.java
package de.thu.recipebook.activities;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import de.thu.recipebook.R;
import de.thu.recipebook.models.Recipe;
import de.thu.recipebook.models.RecipeCollection;

public class RecipeDetailsActivity extends AppCompatActivity {
    private RecipeCollection recipeCollection;
    private Recipe recipe;

    private TextView nameTextView;
    private TextView countryTextView;
    private TextView ingredientsTextView;
    private TextView instructionsTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recipe_details);

        nameTextView = findViewById(R.id.text_view_name);
        countryTextView = findViewById(R.id.text_view_country);
        ingredientsTextView = findViewById(R.id.text_view_ingredients);
        instructionsTextView = findViewById(R.id.text_view_instructions);

        recipeCollection = RecipeCollection.getInstance();
        Recipe nextRecipe = recipeCollection.getNextRecipe(recipe);
        setRecipe(nextRecipe);
    }

    public void nextButtonOnClick(View view) {
        Recipe nextRecipe = recipeCollection.getNextRecipe(recipe);
        setRecipe(nextRecipe);
    }


    public void setRecipe(Recipe recipe) {
        nameTextView.setText(recipe.getName());
        countryTextView.setText(recipe.getCountry());
        ingredientsTextView.setText(recipe.getIngredients());
        instructionsTextView.setText(recipe.getInstructions());

        this.recipe = recipe;
    }
}

File: Exercise 2/Kotlin/src/main/java/de/thu/recipebook/activities/RecipeDetailsActivity.kt
package de.thu.recipebook.activities

import androidx.appcompat.app.AppCompatActivity
import de.thu.recipebook.models.RecipeCollection
import android.widget.TextView
import android.os.Bundle
import android.view.View
import de.thu.recipebook.R
import de.thu.recipebook.models.Recipe

class RecipeDetailsActivity : AppCompatActivity() {
    private var recipeCollection: RecipeCollection? = null
    private var recipe: Recipe? = null

    private var nameTextView: TextView? = null
    private var countryTextView: TextView? = null
    private var ingredientsTextView: TextView? = null
    private var instructionsTextView: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recipe_details)
		
        nameTextView = findViewById(R.id.text_view_name)
        countryTextView = findViewById(R.id.text_view_country)
        ingredientsTextView = findViewById(R.id.text_view_ingredients)
        instructionsTextView = findViewById(R.id.text_view_instructions)

        recipeCollection = RecipeCollection.instance
        val nextRecipe = recipeCollection?.getNextRecipe(recipe)
        if (nextRecipe != null) {
            setRecipe(nextRecipe)
        }
    }

    fun nextButtonOnClick(view: View?) {
        val nextRecipe = recipeCollection!!.getNextRecipe(recipe)
        setRecipe(nextRecipe)
    }

    fun setRecipe(recipe: Recipe) {
        nameTextView!!.text = recipe.name
        countryTextView!!.text = recipe.country
        ingredientsTextView!!.text = recipe.ingredients
        instructionsTextView!!.text = recipe.instructions

        this.recipe = recipe
    }
}